File: device-sdk/src/main/java/io/oddworks/device/metric/OddUserNewMetric.java
package io.oddworks.device.metric;


public class OddUserNewMetric extends OddMetric {
    private static final String TAG = OddUserNewMetric.class.getSimpleName();

    protected static String userNewAction = ACTION_USER_NEW;
    protected static boolean userNewEnabled = false;

    public OddUserNewMetric() {

    }

    public static void setAction(String action) {
        userNewAction = action;
    }

    public static void setEnabled(boolean enabled) {
        userNewEnabled = enabled;
    }

    @Override
    public String getAction() {
        return userNewAction;
    }

    @Override
    public boolean getEnabled() {
        return userNewEnabled;
    }
}

File: device-sdk/src/main/java/io/oddworks/device/metric/OddUserNewMetric.kt
package io.oddworks.device.metric

import org.json.JSONObject


class OddUserNewMetric(contentType: String? = null, contentId: String? = null, meta: JSONObject? = null) : OddMetric(contentType, contentId, meta) {

    override val action: String
        get() = OddUserNewMetric.action

    override val enabled: Boolean
        get() = OddUserNewMetric.enabled

    companion object {
        private val TAG = OddUserNewMetric::class.java.simpleName

        var action = OddMetric.ACTION_USER_NEW
        var enabled = false
    }
}
